common/basecomponent/base.go

```diff
@@ -237,41 +344,79 @@ func setupKafka(cfg *config.Dendrite) (sarama.Consumer, sarama.SyncProducer) {
 
 // setupNaffka creates kafka consumer/producer pair from the config.
 func setupNaffka(cfg *config.Dendrite) (sarama.Consumer, sarama.SyncProducer) {
-	var err error
-	var db *sql.DB
+	var sqlDB *sql.DB
 	var naffkaDB *naffka.DatabaseImpl
-
 	uri, err := url.Parse(string(cfg.Database.Naffka))
-	if err != nil || uri.Scheme == "file" {
-		db, err = sql.Open(common.SQLiteDriverName(), string(cfg.Database.Naffka))
+	if err != nil {
+		panic(err)
+	}
+	switch uri.Scheme {
+	case "file":
+		sqlDB, err = sql.Open(common.SQLiteDriverName(), string(cfg.Database.Naffka))
 		if err != nil {
 			logrus.WithError(err).Panic("Failed to open naffka database")
 		}
 
-		naffkaDB, err = naffka.NewSqliteDatabase(db)
+		naffkaDB, err = naffka.NewSqliteDatabase(sqlDB)
 		if err != nil {
 			logrus.WithError(err).Panic("Failed to setup naffka database")
 		}
-	} else {
-		db, err = sql.Open("postgres", string(cfg.Database.Naffka))
+	case "postgres":
+		fallthrough
+	default:
+		sqlDB, err = sql.Open("postgres", string(cfg.Database.Naffka))
 		if err != nil {
 			logrus.WithError(err).Panic("Failed to open naffka database")
 		}
 
-		naffkaDB, err = naffka.NewPostgresqlDatabase(db)
+		naffkaDB, err = naffka.NewPostgresqlDatabase(sqlDB)
 		if err != nil {
 			logrus.WithError(err).Panic("Failed to setup naffka database")
 		}
 	}
 
-	if naffkaDB == nil {
-		panic("naffka connection string not understood")
-	}
-
 	naff, err := naffka.New(naffkaDB)
 	if err != nil {
 		logrus.WithError(err).Panic("Failed to setup naffka")
 	}
 
 	return naff, naff
 }
+
+type mDNSListener struct {
```

Comment:
This shouldn't live here. Keep it isolated to`/cmd/dendrite-p2p-demo` like what has been done with `dendritejs`.